ee/clickhouse/sql/trends/lifecycle.py

```diff
@@ -1,59 +1,161 @@
 _LIFECYCLE_EVENTS_QUERY = """
-SELECT *, if(base_day = toDateTime('0000-00-00 00:00:00'), 'dormant', if(subsequent_day = base_day + INTERVAL {interval}, 'returning', if(subsequent_day > earliest + INTERVAL {interval}, 'resurrecting', 'new'))) as status
+WITH person_activity_including_previous_period AS (
+    SELECT DISTINCT 
+        person_id, 
+        {trunc_func}(events.timestamp) day 
+
+    FROM events
+        JOIN ({GET_TEAM_PERSON_DISTINCT_IDS}) pdi 
+            ON events.distinct_id = pdi.distinct_id
+
+    WHERE team_id = %(team_id)s AND {event_query} {filters}
+
+    GROUP BY 
+        person_id, 
+        day
+        
+    HAVING 
+        day <= toDateTime(%(date_to)s) 
+        AND day >= toDateTime(%(prev_date_from)s)
+
+), person_activity_as_array AS (
+    SELECT DISTINCT 
+        person_id, 
+        groupArray({trunc_func}(events.timestamp)) day 
```

Comment:
Can we change all "day" references to "period" or something generic so it accounts for week/months/hours

Reply:
have updated, what do you think?